modules/core/src/math/matrix/vpMatrix.cpp

```diff
@@ -1246,14 +1364,10 @@ vpMatrix vpMatrix::operator*(const vpVelocityTwistMatrix &V) const
   vpMatrix M;
   M.resize(rowNum, 6, false, false);
 
-#if defined(VISP_HAVE_LAPACK) && !defined(VISP_HAVE_LAPACK_BUILT_IN)
-  double alpha = 1.0;
-  double beta = 0.0;
-  char trans = 'n';
-
-  vpMatrix::blas_dgemm(trans, trans, V.colNum, rowNum, colNum, alpha, V.data, V.colNum, data, colNum, beta, M.data,
-                       V.colNum);
-#else
+  // Considering perfMatrixMultiplication.cpp benchmark,
```

Comment:
I have a small perf degradation. Before with Intel MKL and with: `./testGenericTracker -d -D`

```
Computation time, Mean: 4.7648 ms ; Median: 4.45264 ms ; Std: 2.96486 ms
Max translation error: 0.000958766
Max thetau error: 0.342824

Computation time, Mean: 4.95577 ms ; Median: 4.4342 ms ; Std: 3.97667 ms
Max translation error: 0.000958766
Max thetau error: 0.342824

Computation time, Mean: 5.00756 ms ; Median: 4.47778 ms ; Std: 4.32292 ms
Max translation error: 0.000958766
Max thetau error: 0.342824
```

After:

```
Computation time, Mean: 5.28233 ms ; Median: 4.78821 ms ; Std: 3.886 ms
Max translation error: 0.000956362
Max thetau error: 0.342824

Computation time, Mean: 5.22407 ms ; Median: 4.7417 ms ; Std: 3.83906 ms
Max translation error: 0.000956362
Max thetau error: 0.342824

Computation time, Mean: 5.09045 ms ; Median: 4.7489 ms ; Std: 3.07056 ms
Max translation error: 0.000956362
Max thetau error: 0.342824
```

With `sampling_step==1`:

Before:

```
Computation time, Mean: 29.1824 ms ; Median: 28.1158 ms ; Std: 10.6638 ms
Max translation error: 0.00110701
Max thetau error: 0.372656

Computation time, Mean: 29.2613 ms ; Median: 27.8002 ms ; Std: 10.6697 ms
Max translation error: 0.00110701
Max thetau error: 0.372656

Computation time, Mean: 29.5986 ms ; Median: 28.3655 ms ; Std: 10.9398 ms
Max translation error: 0.00110701
Max thetau error: 0.372656
```

After:

```
Computation time, Mean: 32.747 ms ; Median: 30.817 ms ; Std: 12.4494 ms
Max translation error: 0.00110701
Max thetau error: 0.372656

Computation time, Mean: 32.6962 ms ; Median: 31.8308 ms ; Std: 12.0067 ms
Max translation error: 0.00110701
Max thetau error: 0.372656

Computation time, Mean: 32.9085 ms ; Median: 31.1582 ms ; Std: 12.3883 ms
Max translation error: 0.00110701
Max thetau error: 0.372656
```

Reply:
I need to check on my side.

Reply:
The table below shows the mean time obtained on my Ununtu laptop:

| testGenericTracker -d -D                                                                       | sample step<br>me=1<br>dense=4 | sample step<br>me=4<br>dense=4 | sample step<br>me=8<br>dense=4 | sample step<br>me=2<br>dense=1<br> |
|------------------------------------------------------------------------------------------------|:------------------------------:|:------------------------------:|:------------------------------:|:----------------------------------:|
| master branch                                                                                  | 16.4681 ms                     | 7.22376 ms                     | 5.60572 ms                     | 40.8489 ms                         |
| With PR #736 and MKL                                                                           | 17.3494 ms                     | 8.10851 ms                     | 5.65789 ms                     | 42.8151 ms                         |
| With PR #736 and MKL plus<br>`mult2Matrices(vpHomogeneousMatrix)`<br>using `MKL::blas_dgemm`                                 | 16.844 ms                      | 7.2696 ms                      | 5.60208 ms                     | 42.3033 ms                         |
| With PR #736 and MKL plus<br>`mult2Matrices(vpHomogeneousMatrix)`<br>`operator*(vpVelocityTwistMatrix)`<br>using `MKL::blas_dgemm` | 16.3295 ms                     | 6.99961 ms                     | 5.42807 ms                     | 39.8107 ms                         |

You are right, I will also introduce `blas_dgemm` in `vpMatrix::mult2Matrices(const vpMatrix &A, const vpMatrix &B, vpHomogeneousMatrix &C)` and `vpMatrix::operator*(const vpVelocityTwistMatrix &V)` since the table shows similar results or a small time improvement (between first and last table lines).

Without using `blas_dgemm` we can see a perf decrease (lines 2 and 3).